design_docs/Configuration.md

```diff
@@ -0,0 +1,54 @@
+# Configuration File
+
+The configuration file is a segmented JSON file, containing groups of key: value pairs.
+
+Each of the labelled groups has a `description` label, giving the context for the contained values. The current config groups are described below.
+
+## `moi_tests`
+
+This section contains relates to the application of Mode Of Inheritance tests. These are not set based on the prevalence
+any individual disease in any specific population. Feedback from clinical teams will be used to further refine these
+settings
+
+* gnomad_dominant - the max gnomad population AF permitted for Mono-allelic disease
+* gnomad_max_homs_dominant - max number of Homs in Gnomad for Mono-allelic disease
+* gnomad_max_homs_recessive - max number of Homs in Gnomad for Bi-allelic disease
+* gnomad_max_homs_recessive - max number of variants in Gnomad for Mono-allelic disease
+* category_2_new_only - analysis flag, if True the category 2 check will only consider where a gene is new in the PanelApp data (i.e. the more granular check of green genes with update MOI is not done)
+
+## `variant_object`
+
+This section relates to the creation of variant objects from CyVCF2 classes. When parsing a VCF using Cyvcf2, the
+variant representation objects use a custom representation of the INFO fields, which cannot be pickled. This limits the
+possible parallelisation of analysis downstream. Extracting out into dictionaries opens up opportunities downstream, and
+these fields are used to set up the dictionary fields.
+
+CSQ_STRING is used in 3 separate places:
+
+1. Hail to construct the CSQ string
+2. BCFTools to write into the header
+3. in downstream analysis to digest the CSQ content
+
+* csq_string - |-delimited series of values, describing the content in the CSQ field
+* var_info_keep - values from the VCF variant INFO field which will be kept. This will be replaced with more generic
+logic, so that the fields don't need to be enumerated in config, which feels brittle.
+
+## `filter`
+
+This section relates to the variant filtration and categorisation process, currently executed in Hail query. These
+default settings are primarily based on a clinical-team provided specification, and can be replaced for a bespoke run.
+
+* af_semi_rare - a minimum AF threshold to be applied to all variants
+* min_samples_to_ac_filter - minimum number of samples in the joint-call in order to apply common-in-cohort filter
+* ac_threshold - where the joint-call filter is applied, remove all variants more common than this in the callset
```

Comment:
"more common than this" is a bit unclear -- what does this threshold represent? Is it a fraction (in [0,1]) of samples that need to have at least one copy of the variant?